pilot/pkg/config/clusterregistry/conversion.go

```diff
@@ -84,6 +84,46 @@ func (cs *ClusterStore) GetPilotClusters() []*k8s_cr.Cluster {
 	return cs.clusters
 }
 
+// ReadClustersV2 reads multiple clusters based upon the label istio/multiCluster
+func ReadClustersV2(k8s kubernetes.Interface, cs *ClusterStore) (errList error) {
+	err := getClustersConfigsV2(k8s, cs)
+	if err != nil {
+		// Errors were encountered, but cluster store was populated
+		log.Errorf("The following errors were encountered during multicluster label processing:",
+			err)
+	}
+
+	return nil
+}
+
+// getClustersConfigsV2 reads mutiple clusters from secrets with labels
+func getClustersConfigsV2(k8s kubernetes.Interface, cs *ClusterStore) (errList error) {
+	listOptions := metav1.ListOptions{ LabelSelector: "istio/multiCluster" }
+        clusterSecrets, err := k8s.CoreV1().Secrets("istio-system").List(listOptions)
+
+	if err != nil {
+		return err
+	}
+	for _, secret := range clusterSecrets.Items {
+		cluster := k8s_cr.Cluster{}
```

Comment:
Why do you add empty struct here? cluster does not have any values set before it gets appended to the slice. Either populate its element with useful info, or just drop it completely since it appears it is not used.

Reply:
I can provide no reasonable explanation other than the code doesn't work without it.  Still, you correct on initial screening, adding to the clusters list an empty cluster object doesn't seem correct.   I'll sort out a correct implementation.

Cheers
-steve

Reply:
The reason adding cluster is needed is in this block of code:
https://github.com/istio/istio/blob/master/pilot/pkg/bootstrap/server.go#L529-L531

Only the cluster name is needed, not the entire k8s_cr object for ConfigV2.  For Config V1, perhaps the k8s_cr object would be useful in the future, but is unused today.